internal/lexer/lexer_test.go

```diff
@@ -164,46 +164,50 @@ func compareMetric(t *testing.T, tests map[string]gostatsd.Metric, namespace str
 
 var parselineBlackhole *gostatsd.Metric
 
-func benchmarkLexer(dp *DatagramParser, input string, b *testing.B) {
+func benchmarkLexer(ns string, input string, b *testing.B) {
 	slice := []byte(input)
 	var r *gostatsd.Metric
-	dp.metricPool = pool.NewMetricPool(0)
+
+	l := &Lexer{
+		MetricPool: pool.NewMetricPool(0),
+	}
+
 	b.ReportAllocs()
 	b.ResetTimer()
 	for n := 0; n < b.N; n++ {
-		r, _, _ = dp.parseLine(slice)
+		r, _, _ = l.Run(slice, ns)
 		r.Done()
 	}
 	parselineBlackhole = r
```

Comment:
One day I'll go back and figure out why I needed this line and comment it :coneofshame:

Something something optimizer.

Reply:
Oh wow, looks like maybe I didn't add it, it's some real dawn of time stuff.  I'm pretty sure that since `r.Done()` was added, it's no longer necessary.